connector.rb

```diff
@@ -1,37 +1,2827 @@
+          field[:type] = :boolean
+          field[:control_type] = :checkbox
+          field[:convert_input] = :boolean_conversion
+          field = field.deep_merge(
+            {
+              type: :boolean,
+              control_type: :checkbox,
+              convert_input: :boolean_conversion,
+              toggle_hint: 'Pick value',
+              toggle_field: {
+                type: :boolean,
+                control_type: :text,
+                convert_input: :boolean_conversion,
+                toggle_hint: 'Provide value'
+              }
+            }
+          )
+        when 'Float', 'MonetaryAmount', 'Decimal'
+          field[:type] = :number
+          field[:control_type] = :number
+          field[:convert_input] = :float_conversion
+        when 'Int'
+          field[:type] = :integer
+          field[:control_type] = :integer
+          field[:convert_input] = :integer_conversion
+        when 'ISO8601Date'
+          field = field.deep_merge(
+            {
+              type: :date,
+              control_type: :date,
+              convert_input: :date_conversion,
+              toggle_hint: 'Pick date',
+              toggle_field: {
+                type: :date,
+                control_type: :text,
+                convert_input: :date_conversion,
+                toggle_hint: 'Provide value'
+              }
+            }
+          )
+        when 'ISO8601DateTime', 'ISO8601Timestamp'
+          field = field.deep_merge(
+            {
+              type: :date_time,
+              control_type: :date_time,
+              convert_input: :render_iso8601_timestamp,
+              toggle_hint: 'Pick date and time',
+              toggle_field: {
+                type: :date_time,
+                control_type: :text,
+                convert_input: :render_iso8601_timestamp,
+                toggle_hint: 'Provide value'
+              }
+            }
+          )
+        else # 'String', 'ID', ...
+          field = field.merge(
+            {
+              type: :string,
+              control_type: :text
+            }
+          )
+        end
+      when 'ENUM'
+        enum_type = call(
+          'get_schema_type',
+          connection,
+          report_problem,
+          'ENUM',
+          type['name']
+        )
+        pick_list = enum_type&.[]('enumValues')&.map do |e|
+          [
+            call('labelize', e['name'], e['isDeprecated']),
+            e['name']
+          ]
+        end
+        field = field.merge(
+          {
+            type: :string,
+            control_type: :select,
+            pick_list: pick_list,
+            toggle_hint: 'Select from list'
+          }
+        )
+        call(
+          'apply_field_description',
+          connection,
+          field,
+          enum_type['description']
+        )
+        field[:toggle_field] = field.merge(
+          {
+            type: :string,
+            control_type: :text,
+            pick_list: nil,
```

Comment:
This was part of the GraphQL connector. Yes, in case the input becomes a string instead of a list it clears the `pick_list` values as there is no need for that list on a `text` control. Same for `optional`, if some decided to not use the list they can send a `nil` value.